mosaic-runtime/src/commonTest/kotlin/com/jakewharton/mosaic/MosaicTest.kt

```diff
@@ -24,4 +29,49 @@ class MosaicTest {
 			""".trimMargin().replaceLineEndingsWithCRLF(),
 		)
 	}
+
+	@Test fun counter() = runTest {
+		val strings = mutableListOf<CharSequence>()
+
+		runMosaic(
+			body = {
+				var count by mutableIntStateOf(0)
+
+				setContent {
+					Text("The count is: $count")
+				}
+
+				for (i in 1..20) {
+					delay(250)
+					count = i
+				}
+			},
+			display = {
+				strings.add(it)
+			},
+		)
+
+		assertThat(strings.size).isEqualTo(22)
+		assertThat(strings[0]).isEqualTo(
+			"""
+			|${ansiBeginSynchronizedUpdate}The count is: 0
+			|$ansiEndSynchronizedUpdate
+			""".trimMargin().replaceLineEndingsWithCRLF(),
+		)
```

Comment:
I added some thoughts at https://github.com/JakeWharton/mosaic/issues/4#issuecomment-2285275775. I don't want to jump the gun on too many of these tests without building a good abstraction for them. We have a pretty good track record of building Turbine-like APIs for this and have it be successful.

Here's one example: https://github.com/cashapp/redwood/blob/831b4c9d7ced73324c6bd8098e770900b9158c39/samples/counter/presenter/src/commonTest/kotlin/com/example/redwood/counter/presenter/CounterTest.kt#L30-L50

Reply:
I completely agree that it is necessary to make an abstraction, I just wanted there to be a minimal simple unit test from which to build further.